Set up registration endpoint to allow more people to submit extractors

mfrdrcks opened the pull request:
Filter the default list of extractors by approved.
Add a property for allowing anonymous submission.
Set up a registration endpoint that will redirect to a given URL, or give a 501 error if no external URL is provided.
Link to the registration endpoint on the login page.

Currently there is no mechanism to limit how many extractions may be submitted. If that is a problem, I can look at how to do that either as part of this PR, or as a new PR.
I did also verify that the transformations page does work, so the short form where you submit the json for the converter/extractor will work for the world. I did not error check what happens if not all the needed information is added to the form.
If the config.json (or config.py if still using that) does not contain a "REGISTRATION_URL" property with a link to a registration link then a 501 error will be returned that states that local registration is not implemented. If that property is present then the registration url will be redirected to from the registration endpoint.
There is an additional recognized property to allow anonymous submissions that defaults to false.

mfrdrcks committed:
Filter the default list of extractors by approved.
Add a property for allowing anonymous submission.
Set up a registration endpoint that will redirect to a given URL, or give a 501 error if no external URL is provided.
Link to the registration endpoint on the login page.

mfrdrcks committed:
Update README.md

Add formatting to Configuration Settings.
Include ANONYMOUS_SUBMISSION to the Configuration Settings.

mfrdrcks committed:
Added a docker-compose for ease of testing.
Added a section to the README about using the docker-compose.
Added a config.json that is set for easy use of docker-compose in a non-secure manner.

mfrdrcks committed:
Merge branch 'feature/add-registration' into feature/docker-compose-for-easy-local-launch

mfrdrcks committed:
Update image name.

lmarini commented on transformations/pages.py:
```diff
@@ -64,8 +64,8 @@ def softwares():
-    # Check if user has logged in
-    if g.user is not None:
+    # Check if user has logged in or anonymous submission is allowed
+    if g.user is not None or current_app.config["ANONYMOUS_SUBMISSION"] is "True":
```
I think the second statement needs to be `==` instead of `is` since we are doing string equality. Or you can cast the return of config to a boolean and compare it to `True`.

mfrdrcks committed:
Readd instance/config.json to the .gitignore

mfrdrcks committed:
Change to do proper string comparison

Added lower() to the string to allow any capitalization of true in the configuration file.

lmarini committed:
Merge pull request #4 from clowder-framework/feature/docker-compose-for-easy-local-launch

docker-compose for testing
